### PR TITLE
PLT-7742: Add confirm modal to data retention settings.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Add a confirm modal to the data retention settings page.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7742

#### Checklist
- [x] Has UI changes
- [ ] Includes text changes and localization file updates

### CODE REVIEW

**Comment by grundleborg** [comment]
@jasonblais I've deviated a bit from the design in the ticket - let me know what you think, and any changes you want.

File: components/admin_console/data_retention_settings.jsx [hunk 1]
```diff
@@ -38,10 +41,124 @@ export default class DataRetentionSettings extends AdminSettings {
...
+    };
+
+    handleSaveCanceled = () => {
+        this.setState({showConfirmModal: false});
+    };
+
+    renderConfirmModal() {
+        const title = (
+            <FormattedMessage
+                id='admin.data_retention.confirmChangesModal.title'
```
Comment (cpanato) [inline components/admin_console/data_retention_settings.jsx]: missing the string in the en.json file

Comment (grundleborg) [reply components/admin_console/data_retention_settings.jsx]: fixed.

Comment (cpanato) [reply components/admin_console/data_retention_settings.jsx]: cool, thanks! 😄

**Pull request merged**